test/controllers/sessions_controller_test.rb

```diff
@@ -46,15 +46,32 @@ class SessionsControllerTest < ActionController::TestCase
     assert_equal @user.id, current_user.id
   end
 
-  test 'should not create a new session' do
+  test 'should not create a new session with correct user_name' do
     Ldap.default.destroy!
 
     assert_no_difference '@user.logins.count' do
-      post :create, params: { username: @user.email, password: 'wrong' }
+      assert_difference 'Fail.all.count' do
+        post :create, params: { username: @user.email, password: 'wrong' }
+      end
     end
 
     assert_response :success
     assert_nil current_user
+    assert_equal 1, Fail.where(user_id: @user.id).count
+  end
+
+  test 'should not create a new session with incorrect user_name' do
+    Ldap.default.destroy!
+
+    assert_no_difference '@user.logins.count' do
+      assert_difference 'Fail.all.count' do
+        post :create, params: { username: 'not_exist', password: 'wrong' }
+      end
+    end
+
+    assert_response :success
+    assert_nil current_user
+    assert Fail.all.all? { |fail| fail.user.blank? }
```

Comment:
```suggestion
    assert Fail.last.user.blank?
```
Esto por si después agregamos en los fixtures algo asociado a un usuario, acá nos va a fallar.

Reply:
claro. Perfecto.